src/main/webapp/profLoader.js

```diff
@@ -11,14 +11,30 @@
   const profKey = urlParams.get("prof-key");
   const profName = urlParams.get("prof-name");
 
+  function fillTitle(profData) {
+    document.getElementById("prof-name").innerHTML = profName;
+    linkContainer = document.getElementById("link-container");
+
+    for (let dataHolder of profData) {
+      const newLink = document.createElement("a");
+      newLink.innerHTML = `${dataHolder.course} - ${dataHolder.term}`;
```

Comment:
FYI this would be trivially easy to execute a XSS attack with. They could override your variable here in dev tools and insert their own script tag which would then run with full page & origin permissions.

Not a concern for this app, but good to know if you're ever making a production website.